### PR TITLE
Desktop : Enable external editing in viewer mode (Fixes #2762)

### LINKED ISSUE #2762: When the editor is in Viewer mode, enable "Open in external editor" menu item
This item was recently disabled in a pull request, but it should always be enabled in all views.

Forum thread: https://discourse.joplinapp.org/t/external-editor-shortcut-available-in-any-layout-view-mode/6958?u=laurent

### PR DESCRIPTION
**Pull request by anihm136**
Removed disable of "Edit in external editor", "Copy", "Select All" and "Search in Current Note" features in viewer mode, as these features are valid for all modes. These options have been removed from the enable validation in `app.js`.

Fixes #2762

### CODE REVIEW

**Comment by CalebJohn** [comment]
Hi @anihm136 

It looks like you've added files to the gitignore that aren't related to this project (editor specific files). These files should instead be added to your own [global gitignore](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c), this will help you so that you don't have to add these to the gitignore of every project you use.

File: ElectronClient/app.js [hunk 1]
```diff
@@ -1188,7 +1188,7 @@ class Application extends BaseApplication {
 		const selectedNoteIds = state.selectedNoteIds;
 		const note = selectedNoteIds.length === 1 ? await Note.load(selectedNoteIds[0]) : null;
 
-		for (const itemId of ['copy', 'paste', 'cut', 'selectAll', 'bold', 'italic', 'link', 'code', 'insertDateTime', 'commandStartExternalEditing', 'showLocalSearch']) {
+		for (const itemId of ['paste', 'cut', 'bold', 'italic', 'link', 'code', 'insertDateTime']) {
```
Comment (tessus) [inline ElectronClient/app.js]: Why did you remove more than the asked command?

File: .gitignore [hunk 2]
```diff
@@ -48,6 +48,11 @@ joplin-webclipper-source.zip
 Tools/commit_hook.txt
 .vscode/*
 *.map
+GTAGS
+GRTAGS
+GPATH
+cscope.out
+.starscope.db
```
Comment (tessus) [inline .gitignore]: remove

Comment (anihm136) [reply .gitignore]: Done

File: ElectronClient/app.js (hunk 1 above)
Comment (anihm136) [reply ElectronClient/app.js]: I was checking which of the functions worked in viewer mode, as I did not know why they are being disabled. Missed reverting that. Sorry

**Comment by anihm136** [comment]
> Hi @anihm136
> 
> It looks like you've added files to the gitignore that aren't related to this project (editor specific files). These files should instead be added to your own [global gitignore](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c), this will help you so that you don't have to add these to the gitignore of every project you use.

Ah thanks. I did not know about the global `.gitignore`, and since there was a vscode ignore pattern there I thought that was the practice. Good learning for me

**Comment by tessus** [comment]
This one is actually a duplicate of #2763, but this implemention looks cleaner.

Laurent has to decide which one is merged.

**Comment by laurent22** [comment]
See https://github.com/laurent22/joplin/pull/2763#issuecomment-602041823
